modules/integration/tests-integration/tests-lifecycle/src/test/java/org/wso2/carbon/registry/lifecycle/test/LCCheckpointNotificationTestCase.java

```diff
@@ -0,0 +1,433 @@
+        String mediaType = "application/vnd.wso2-service+xml";
+        String description = "This is a test service";
+        String destinationPath = GREGTestConstants.GOVERNANCE_LOCATION + "/service";
+        resourceAdminServiceClient.addResource(destinationPath, mediaType, description, dataHandler);
+        org.wso2.carbon.registry.resource.stub.common.xsd.ResourceData[] data = resourceAdminServiceClient
+                .getResource(absPath);
+        assertNotNull(data, "Service not found");
+    }
+
+    /**
+     * This method act as the test case to subscribe for Checkpoint Notification.
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.greg", description = "Subscribe LC Approval Needed notification while state change",
+            dependsOnMethods = "testAddLcToService")
+    public void testSubscribeLCCheckpointNotification() throws Exception {
+        addRole();
+        assertTrue(consoleNotificationSubscribe(absPath, "CheckpointNotification"));
+    }
+
+    /**
+     * This method act as the test case to create a service.
+     *
+     * @throws RegistryException
+     * @throws CustomLifecyclesChecklistAdminServiceExceptionException
+     * @throws RemoteException
+     */
+    @Test(groups = "wso2.greg", description = "Add lifecycle to a service", dependsOnMethods = "testCreateService")
+    public void testAddLcToService()
+            throws RegistryException, CustomLifecyclesChecklistAdminServiceExceptionException, RemoteException {
+
+        wsRegistryServiceClient.associateAspect(absPath, aspectName);
+        lifeCycle = lifeCycleAdminServiceClient.getLifecycleBean(absPath);
+        Property[] properties = lifeCycle.getLifecycleProperties();
+        boolean lifecycleStatus = false;
+        for (Property prop : properties) {
+            if (prop.getKey().contains(aspectName)) {
+                lifecycleStatus = true;
+            }
+        }
+        assertTrue(lifecycleStatus, "LifeCycle not added to service");
+    }
+
+    /**
+     * This method is used to subscribe for a notification.
+     *
+     * @param path                  registry path of the resource.
+     * @param eventType             event type to subscribe.
+     * @return                      true if subscription is done.
+     * @throws RemoteException
+     * @throws RegistryException
+     */
+    private boolean consoleNotificationSubscribe(String path, String eventType)
+            throws RemoteException, RegistryException {
+        // subscribe for management console notifications
+        SubscriptionBean bean = infoServiceAdminClient.subscribe(path, "work://RoleSubscriptionTest", eventType,
+                sessionCookie);
+        return bean.getSubscriptionInstances() != null;
+    }
+
+    /**
+     * This method used to add a role
+     *
+     * @return              true if the role is successfully added.
+     * @throws Exception
+     */
+    private boolean addRole() throws Exception {
```

Comment:
Thanks for the comment. Exception is thrown using "Exception" class here because userManagementClient.roleNameExists() method used in this method throws the exception using "Exception" class. Added a public JIRA[1] to refactor userManagementClient class excetion class. I'll change this exception handling after completion of the code refactoring in userManagementClient class.

[1] - https://wso2.org/jira/browse/REGISTRY-2463